Adding a `from` argument to render_preview to specify the preview class rather than inferring it (#1572)
ViewComponent/view_component: docs/guide/previews.md

end
```

> By default, the name of the preview to be rendered is inferred from the name of the current test file. Use `from` to explicitly set the preview to be rendered:

Reviewer: ```suggestion
By default, the preview class is inferred from the name of the current test file. Use `from` to set it explicitly:
```

```ruby
class ExampleTest < ViewComponent::TestCase
  def test_render_preview

>     render_preview(:with_default_title, klass: ExampleComponentPreview)

Reviewer 2: ```suggestion
    render_preview(:with_default_title, preview_class: ExampleComponentPreview)
```

Minor API suggestion, thoughts on naming the argument `preview_class` since we should be able to use `class` with a prefix?

Reviewer: @BlakeWilliams that's how @jonrohan implemented it originally, but I've been of the general belief that ruby param names should not be duplicative of the method's name. Since the method name already includes `preview`, I think `klass` is more idiomatic here. 

But I don't feel too strongly either way. I'd be happy to go with `preview_class` if you feel strongly about it ❤️

Reviewer 2: > Since the method name already includes `preview`, I think `klass` is more idiomatic here.

Yeah, I generally agree about not "stuttering" as Go would call it (preview preview). I generally try to avoid `klass` though, since I've never found it to be descriptive enough and the spelling throws me off.

Is there an alternative name that would give us the best of both worlds? `component_class` maybe?

Reviewer: Hmm. It isn't a component class though, it's a preview class.

Reviewer 3: Maybe something simpler such as `from` or `in`? This way it reads like a sentence: Render preview_method from/in PreviewClass.

Reviewer: @Spone lovely. I'd 👍🏻 `from`.

Reviewer 4: do we have sorbet or any other strong typing in this gem? the naming is useful, but more specifically stating that the param must be a `ViewComponent::Preview` subclass would be 100x more useful

i also like the way this looks with `from`

```ruby
render_preview(:default, from: Primer::Beta::MyComponentPreview)
```

this reads really well, IMO 👍🏻
